lib/corebulk.go

```diff
@@ -158,12 +158,21 @@ func (b *BulkIndexer) Start() {
 		}
 		// Backwards compatibility
 		b.startHttpSender()
-		b.startDocChannel()
+
+		docChannelStopWait := &sync.WaitGroup{}
+		docChannelStopWait.Add(1)
+		b.startDocChannel(docChannelStopWait)
```

Comment:
An alternative formulation might be to have startDocChannel return a channel (or a waitgroup) that we wait on.  I generally favor channels for this sort of thing (they are directional, which tends to make the usage more obvious, for example).

Reply:
Ah, I like that, I'll give it a try here